### PR TITLE
Bug 1726455: Reconcile container ResourceRequirements in resource merge

### PR DESCRIPTION
**Pull request by jcpowermac**
- Add ensureResourceRequirements function
If existing and required are not equal will DeepCopy required
into existing.

- Update core_test to include testing container resources limits addition
and removal.

### CODE REVIEW

File: lib/resourcemerge/core.go [hunk 1]
```diff
@@ -445,6 +446,13 @@ func ensureSELinuxOptions(modified *bool, existing *corev1.SELinuxOptions, requi
 	setStringIfSet(modified, &existing.Level, required.Level)
 }
 
+func ensureResourceRequirements(modified *bool, existing *corev1.ResourceRequirements, required corev1.ResourceRequirements) {
```
Comment (abhinavdahiya) [inline lib/resourcemerge/core.go]: include both https://godoc.org/k8s.io/api/core/v1#ResourceRequirements `Limits` and `Requests`

Comment (jcpowermac) [reply lib/resourcemerge/core.go]: @abhinavdahiya ok will update.

File: lib/resourcemerge/core_test.go [hunk 2]
```diff
@@ -113,6 +114,100 @@ func TestEnsurePodSpec(t *testing.T) {
 				Containers: []corev1.Container{
 					corev1.Container{Name: "test"}}},
 		},
+		{
+			name: "remove limits and requests on container",
```
Comment (abhinavdahiya) [inline lib/resourcemerge/core_test.go]: nit: add a no-op ?

File: lib/resourcemerge/core_test.go [hunk 3]
```diff
@@ -113,6 +114,138 @@ func TestEnsurePodSpec(t *testing.T) {
...
+						Resources: corev1.ResourceRequirements{
+							Limits:   corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("4m")},
+							Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("4m")},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "match limits and requests on container",
```
Comment (jcpowermac) [inline lib/resourcemerge/core_test.go]: @abhinavdahiya added matching existing and input last night with `expectedModified: false`

**Pull request merged**